src/App.tsx

```diff
@@ -1,12 +1,77 @@
 import React from 'react';
 import './App.scss';
-
-// import { getAll, get5First, getRed } from './api/goods';
+import { GoodsList } from './components/GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  showAll = async () => {
+    const allGoods = await getAll();
+
+    this.setState({
+      goods: allGoods,
+    });
+  };
+
+  showFirstFiveGoods = async () => {
+    const firstFiveGoods = await get5First();
+
+    this.setState({
+      goods: firstFiveGoods,
+    });
+  };
+
+  showRedGoods = async () => {
+    const redGoods = await getRedGoods();
+
+    this.setState({
+      goods: redGoods,
+    });
+  };
```

Comment:
you can create 1 common function for getting goods, and just pass the needed function on the argument when press button